### DISCUSSION TITLE
How to make doors & trapdoors functional

### DISCUSSION THREAD

**Discussion by KianBrose** [Minestom/Minestom#1813] [Help] [answered]
As the title says, how do I in practice make doors and trapdoors functional on a server-side level? I could not find out how to do it through the documentation on https://wiki.minestom.net/

**Comment by emortaldev** [comment, marked as answer]
Check for a player interacting with it via `PlayerBlockInteractEvent`, then invert the open property on the block with 
```java
Block newBlock = event.block.setProperty("open", event.block.getProperty("open") == "true" ? "false" : "true"); // Invert property
event.getInstance().setBlock(event.blockPosition, newBlock); // Set this new block
```
To handle doors which are actually two blocks, you can use `getProperty("half")` which has values `upper` and `lower` to help you find out where the other door block is.
